src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectGenerator.cs

```diff
@@ -486,21 +486,48 @@ public bool TryGetTemplateFromConfigInfo(IFileSystemInfo templateFileConfig, out
             }
         }
 
+        private static IVariableCollection SetupVariables(IEngineEnvironmentSettings environmentSettings, IParameterSet parameters, IVariableConfig variableConfig)
+        {
+            IVariableCollection variables = VariableCollection.SetupVariables(environmentSettings, parameters, variableConfig);
+
+            foreach (Parameter param in parameters.ParameterDefinitions.OfType<Parameter>())
+            {
+                // Add choice values to variables - to allow them to be recognizable unquoted
+                if (param.EnableQuotelessLiterals && param.IsChoice())
+                {
+                    foreach (string choiceKey in param.Choices.Keys)
+                    {
+                        if (
+                            variables.TryGetValue(choiceKey, out object existingValueObj) &&
+                            existingValueObj is string existingValue &&
+                            !string.Equals(choiceKey, existingValue, StringComparison.CurrentCulture)
+                        )
+                        {
+                            throw new InvalidOperationException(string.Format(LocalizableStrings.RunnableProjectGenerator_CannotAddImplicitChoice, choiceKey, existingValue));
+                        }
+                        variables[choiceKey] = choiceKey;
+                    }
+                }
+            }
```

Comment:
would it be better to have this logic in `VariableCollection.SetupVariables`? 

Also it seems `ITemplateParameter` is still used in `Core-RunnableProjects` via `IParameterSet` in `SetupVariables`.